tests/e2e/scale/test_pvc_creation_deletion_scale.py

```diff
@@ -118,10 +110,24 @@ def test_multiple_pvc_creation_deletion_scale(self, namespace, access_mode, inte
             f"Create data present in {log_path}-creation-time.csv file"
         )
 
-        # Delete PVC
-        for obj in pvc_objs:
-            obj.delete()
-            obj.ocp.wait_for_delete(obj.name)
+        # Get pv_name, require pv_name to fetch deletion time data from log
+        # TODO: Revisit on changing below code without threads, for now it looks good
+        # TODO: but run-ci memory growth is increasing ~0.4G with below threads
+        pvc_objs = pvc.get_all_pvc_objs(namespace=self.namespace)
+        threads, pv_name_list = ([] for i in range(2))
+        for pvc_obj in pvc_objs:
```

Comment:
wouldn't we be creating 1500 PVC objects once again here? maybe there is a better way? if the intention is just to create another list with PV names, we should consider a different approach here.

Reply:
Yes, it will be. The problem here is with the help of kube_job object we get info about the PVC (the details we get from oc get pvc <pvc_name> -o yaml), if we want information like what is the backed pv name we don't have way to fetch that using kube_job.
We need pv_name especially to fetch deletion time from the CSI logs, PVC name is not present in delete log.
Definitely need to revisit this piece of code, and during execution observed 0.4G of memory is increasing when executing this lines of code.

Reply:
I just checked in kube_job obj for one PVC, there is '`volumeName': 'pvc-5d944cd5-1bbd-44ed-b548-5fe4e71dd012'`, let me do some validation either this helps to solve the problem.

`{'apiVersion': 'v1', 'items': [{'apiVersion': 'v1', 'kind': 'PersistentVolumeClaim', 'metadata': {'annotations': {'pv.kubernetes.io/bind-completed': 'yes', 'pv.kubernetes.io/bound-by-controller': 'yes', 'volume.beta.kubernetes.io/storage-provisioner': 'openshift-storage.rbd.csi.ceph.com'}, 'creationTimestamp': '2020-10-06T07:03:09Z', 'finalizers': ['kubernetes.io/pvc-protection'], 'managedFields': [{'apiVersion': 'v1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:metadata': {'f:annotations': {'.': {}, 'f:pv.kubernetes.io/bind-completed': {}, 'f:pv.kubernetes.io/bound-by-controller': {}, 'f:volume.beta.kubernetes.io/storage-provisioner': {}}}, 'f:spec': {'f:volumeName': {}}, 'f:status': {'f:accessModes': {}, 'f:capacity': {'.': {}, 'f:storage': {}}, 'f:phase': {}}}, 'manager': 'kube-controller-manager', 'operation': 'Update', 'time': '2020-10-06T07:03:09Z'}, {'apiVersion': 'v1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:spec': {'f:accessModes': {}, 'f:resources': {'f:requests': {'.': {}, 'f:storage': {}}}, 'f:storageClassName': {}, 'f:volumeMode': {}}}, 'manager': 'kubectl-create', 'operation': 'Update', 'time': '2020-10-06T07:03:09Z'}], 'name': 'pvc-test-2ecae34816b043dca4652e02380e8c0a', 'namespace': 'namespace-test-8fe64ac2009347bfb82ff99a6191e7ad', 'resourceVersion': '1728596', 'selfLink': '/api/v1/namespaces/namespace-test-8fe64ac2009347bfb82ff99a6191e7ad/persistentvolumeclaims/pvc-test-2ecae34816b043dca4652e02380e8c0a', 'uid': '5d944cd5-1bbd-44ed-b548-5fe4e71dd012'}, 'spec': {'accessModes': ['ReadWriteOnce'], 'resources': {'requests': {'storage': '25Gi'}}, 'storageClassName': 'ocs-storagecluster-ceph-rbd', 'volumeMode': 'Filesystem', 'volumeName': 'pvc-5d944cd5-1bbd-44ed-b548-5fe4e71dd012'}, 'status': {'accessModes': ['ReadWriteOnce'], 'capacity': {'storage': '25Gi'}, 'phase': 'Bound'}},`

Reply:
above pv names works, will send the fix soon.